lazyfs/src/main.cpp

```diff
@@ -28,30 +28,46 @@ std::thread faults_handler_thread;
 LazyFS fs;
 
 void fht_worker (LazyFS* filesystem) {
-
-    int fd;
-    if ((fd = open (std_config.FIFO_PATH.c_str (), O_RDWR)) < 0) {
+    int fd_fifo, fd_fifo_completed;
+    
+    fd_fifo = open (std_config.FIFO_PATH.c_str (), O_RDWR);
+    if (fd_fifo < 0) {
         spdlog::critical ("[lazyfs.fifo]: failed to open fifo '{}' (error: {})",
                           std_config.FIFO_PATH.c_str (),
                           strerror (errno));
         return;
     }
 
+    bool write_completed_faults = (std_config.FIFO_PATH_COMPLETED != "");
+
+    if (write_completed_faults) {
+        fd_fifo_completed = open (std_config.FIFO_PATH_COMPLETED.c_str (), O_WRONLY);
+        if (fd_fifo_completed < 0) {
+            spdlog::critical ("[lazyfs.fifo]: failed to open fifo '{}' (error: {})",
+                            std_config.FIFO_PATH_COMPLETED.c_str (),
+                            strerror (errno));
+            return;
+        }
+    }
+
     spdlog::info ("[lazyfs.faults.worker]: waiting for fault commands...");
 
     char buffer[MAX_READ_CHUNK];
-
     int ret;
-
     while (true) {
-        if ((ret = read (fd, &buffer, MAX_READ_CHUNK)) > 0) {
+        if ((ret = read (fd_fifo, &buffer, MAX_READ_CHUNK)) > 0) {
 
             buffer[ret - 1] = '\0';
 
             if (!strcmp (buffer, "lazyfs::clear-cache")) {
 
                 spdlog::info ("[lazyfs.faults.worker]: received '{}'", string (buffer));
                 filesystem->command_fault_clear_cache ();
+                
+                if (write_completed_faults) {
+                    const char* clear_cache = "clear-cache";
```

Comment:
```suggestion
                    const char* clear_cache = "finished::clear-cache\n";
```

I the future, we could add fault IDs and/or timestamps to know which `lazyfs::clear-cache` commands are tied to a termination indication in the `fifo_path_completed`. 

Also, do you think we need to synchronize these requests to prevent concurrent writes to the faults FIFO?

Reply:
I added a lock as you suggested